spring-cloud-app-broker-core/src/test/java/org/springframework/cloud/appbroker/manager/BackingAppManagementServiceTest.java

```diff
@@ -135,75 +125,35 @@ void stopApplicationsWithEmptyApplications() {
 			brokeredServicesNoApps,
 			targetService);
 
-		given(appDeployer.getServiceInstance(any(GetServiceInstanceRequest.class)))
-			.willReturn(Mono.just(GetServiceInstanceResponse.builder()
-				.name("foo-service")
-				.plan("plan1")
-				.service("service1")
-				.build()));
-
 		given(targetService.addToBackingApplications(eq(emptyBackingApps), any(), eq("foo-service-id")))
 			.willReturn(Mono.just(emptyBackingApps));
 
-		StepVerifier.create(backingAppManagementService.stop("foo-service-id"))
+		StepVerifier.create(backingAppManagementService.stop("foo-service-id", "service1", "plan1"))
 			.verifyComplete();
 
-		verify(appDeployer).getServiceInstance(any(GetServiceInstanceRequest.class));
 		verify(targetService).addToBackingApplications(eq(emptyBackingApps), any(), eq("foo-service-id"));
 		verifyNoInteractions(managementClient);
 		verifyNoMoreInteractions(appDeployer, targetService, managementClient);
 	}
 
-	@Test
-	void stopApplicationsServiceNotFound() {
```

Comment:
Is there not any value in these negative test cases?

Reply:
Maybe in a different form. Without making a call to CAPI, broker doesn't know if an SI exists or not. So if a management operation invoked for a non-existing service instance, the operation will fail with either non-existing space exception (wtih space-per-si target) or non-existing application instance (with space guid app suffix target). But in any case App Broker will just relay CAPI exception to a caller, so not sure how much value is in testing this. 

Another option I can think of is handling non-existing space / app exception and wrapping that to `ServiceInstanceOperationException` or something. But even then, App Broker can only _guess_ that if a space is not found then SI doesn't exist, there could be other reasons as well. 

So I'm happy to bring this negative test cases back, but I'm not sure what exactly they suppose to be testing. WDYT?